### ISSUE TITLE
failing test (occasionally) 

### ISSUE THREAD

**Issue by Garonenur** [genometools/genometools#797]
I am not sure if it is always in gt repfind, but this test just failed on travis for me with the following (not helpful) error:
```
1620: gt repfind extend self vs query                             : failed

FAILING test, output of testsuite/stest_testsuite/test1620/stderr_7

FAILING test, output of stest_error

Test 1620 'gt repfind extend self vs query': failed:

unexpected return code: 1 != 0:

/home/travis/build/genometools/genometools/testsuite/stest.rb:286:in `run'

/home/travis/build/genometools/genometools/testsuite/stest.rb:449:in `run'

/home/travis/build/genometools/genometools/testsuite/gt_repfind_include.rb:79:in `crosstest'

/home/travis/build/genometools/genometools/testsuite/gt_repfind_include.rb:200:in `block (3 levels) in <top (required)>'

/home/travis/build/genometools/genometools/testsuite/gt_repfind_include.rb:197:in `each'

/home/travis/build/genometools/genometools/testsuite/gt_repfind_include.rb:197:in `block (2 levels) in <top (required)>'

/home/travis/build/genometools/genometools/testsuite/gt_repfind_include.rb:192:in `times'

/home/travis/build/genometools/genometools/testsuite/gt_repfind_include.rb:192:in `block in <top (required)>'

/home/travis/build/genometools/genometools/testsuite/stest.rb:176:in `block in test'

/home/travis/build/genometools/genometools/testsuite/stest.rb:165:in `fork'

/home/travis/build/genometools/genometools/testsuite/stest.rb:165:in `test'

/home/travis/build/genometools/genometools/testsuite/stest.rb:120:in `process'

/home/travis/build/genometools/genometools/testsuite/stest.rb:34:in `block (2 levels) in run'

     [ problem: unexpected return code: 1 != 0

       in: /home/travis/build/genometools/genometools/testsuite/stest_testsuite/test1620 ]
```

I think I saw repfind errors fail a few times and looking at the code of the test I am not sure why they would fail only sometimes. Probably race conditions, but it was the grep that failed and the file it was supposed to process was touched before, so it existed.
Also, the tests in gt_repfind_include around lines 200ff are repeated 3 times without a visible reason?
Could someone who has worked on that code have a look? Maybe they understand where there could be a problem with parallel test execution.

**Comment by satta** [comment]
I guess the test is repeated because it's randomised... so you actually get three test runs with different data. That would also explain the lack of consistency in failing -- some inputs cause the tests to fail. I don't immediately see a problem with parallel execution there.
Difficult to reproduce and debug though, as the code that generates the random input does (currently) not use the global seed set by the test environment. After a quick look at the code, it should be doable to use the global seed to generate new deterministic seeds for `scripts/gen-randseq.rb` (it accepts a `--seed` parameter) and try to find out what kind of input triggers the problem. Any comments?

**Comment by Garonenur** [comment]
getting at least a printout of the seed, so one could rerun that exact failing test would be a requirement for any test that uses random data IMO. Tests only make sens if you can use them to debug. If it just tells us our software is broken without any other info it is useless.

**Comment by satta** [comment]
Indeed. Thanks for having a look. Do you want to make the test reproducible?

**Comment by Garonenur** [comment]
give me a few days, then yes, I'll have a look.

**Comment by satta** [comment]
This seems to have been fixed in #801. Can we close this issue?

**Comment by Garonenur** [comment]
if there is still a problem, it will pop up again.
